system/Validation/Rules.php

```diff
@@ -378,18 +378,15 @@ public function required_with($str = null, string $fields, array $data): bool
 
 		foreach ($fields as $field)
 		{
-			if (array_key_exists($field, $data))
+			if (
+				(strpos($field, '.') !== false && ! empty(dot_array_search($field, $data))) ||
+				(array_key_exists($field, $data) && ! empty($data[$field]))
```

Comment:
I think flipping the order of these two will eek out a little more performance since `empty` and `array_key_exists` and almost assuredly faster than `dot_array_search`.

Reply:
Done! (And for what it’s worth, I tested it.)